### PR TITLE
MAB-321 Add word count validator for tinymce questions in surveyjs

### PR DESCRIPTION
**Pull request by ronnyReliefapp**
# Description

This PR introduces a configurable word limit feature for the custom TinyMCE Editor question type in SurveyJS.

Key implementation details:

- Property Grid: Added a maxWords property to the Editor question settings (default: -1, meaning unlimited).
- Smart Counting Logic: Implemented a  word counting algorithm that correctly handles HTML structure:
  - Block tags (e.g., p, div, br) are treated as spaces to ensure separate paragraphs are counted as separate words.
  - Inline tags (e.g., b, span) are stripped to ensure formatted words (e.g., h<b>b</b>h) count as a single word.
  - HTML Entities (&nbsp) are normalized to spaces.
- Visual Feedback: Added a word counter below the editor that updates as the user types.
- Validation: Integrated with SurveyJS validation lifecycle to prevent form submission if the word count exceeds the limit.

## Useful links

Ticket: https://www.notion.so/cb32a703a24e4e03a3ec47bd6b38c4a4?v=467317821cf24db6b43c59325335e46b&p=23fd463fd8c480c98edbd7f29ece23df&pm=s

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] Improvement (refactor or addition to existing functionality)

# How Has This Been Tested?

I have manually tested this component in the Form Builder, Preview mode, and Display mode.

Test Configuration (Form Builder):

- Added an "Editor" question to the form.
- Navigated to the "General" category in properties.
- Verified maxWords defaults to -1.
- Changed maxWords to 10.

Test Validation & UI (Preview Mode):

- Typed normal text. Verified counter shows "X / 10 words".
- Exceeded 10 words. Verified counter text turned red (e.g., "15 / 10 words (5 over limit)").
- Attempted to complete the survey. Verified that SurveyJS threw a validation error preventing submission.

Test HTML/Formatting Edge Cases:

- Block Tags: Typed "Hello" -> hit Enter -> typed "World". Verified this counts as 2 words (Block tags replaced with space).
- Inline Tags: Typed "hbh" (intercalated bold). Verified this counts as 1 word (Inline tags stripped completely).
- Spaces: Typed "Hello     World". Verified this counts as 2 words (Entities and multiple spaces collapsed).

## Screenshots

Screenshot from 2026-01-08 14-18-40

# Checklist:

( * == Mandatory ) 

- [x] * I have set myself as assignee of the pull request
- [x] * My code follows the style guidelines of this project
- [x] * Linting does not generate new warnings
- [x] * I have performed a self-review of my own code
- [x] * I have put the ticket for review, adding the oort-frontend team to the list of reviewers
- [x] * I have commented my code, particularly in hard-to-understand areas
- [x] * I have put JSDoc comment in all required places
- [x] * My changes generate no new warnings
- [x] * I have included screenshots describing my changes if relevant
- [x] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# More explanation
https://www.loom.com/share/05a716d61b9744faaf51fb304c21d1e5?sid=f87cf896-582a-4f76-93ae-8ceed801b145

### CODE REVIEW

File: libs/shared/src/lib/survey/components/editor.ts [hunk 1]
```diff
@@ -115,6 +146,19 @@ export const init = (
...
+            const wordCount = calculateWordCount(html);
+            if (wordCount > question.maxWords) {
+              question.addError(
+                new CustomError(
+                  `Maximum word limit is ${question.maxWords}. Current: ${wordCount}`,
+                  question
+                )
+              );
+            } else {
+              question.clearErrors();
```
Comment (AntoineRelief) [inline libs/shared/src/lib/survey/components/editor.ts]: @ronnyCapriles this may be problematic if the question has other errors

File: libs/shared/src/lib/survey/components/editor.ts [hunk 2]
```diff
@@ -12,6 +13,29 @@ import { FIELD_EDITOR_CONFIG } from '../../const/tinymce.const';
...
 
+/**
+ * Helper function to count words consistently
+ * 1. Replaces Block tags (p, div, br, etc) with SPACE.
+ * 2. Replaces Inline tags (strong, span, etc) with EMPTY STRING.
+ * 3. Handles &nbsp;
+ *
+ * @param html
+ */
+const calculateWordCount = (html: string): number => {
```
Comment (AntoineRelief) [inline libs/shared/src/lib/survey/components/editor.ts]: @ronnyReliefapp 
There is an error in the linting:

CleanShot 2026-01-09 at 14 21 39


make sure you have the VSCode extension installed for eslint please
+ there should be a comment that runs everytime you try to do a commit and / or push, with the warning

you must fix that warning anytime you see it before moving PR to review

File: libs/shared/src/lib/survey/components/editor.ts [hunk 3]
```diff
@@ -115,6 +146,19 @@ export const init = (
           if (isNil(html)) {
             return;
           }
+          if (question.maxWords && question.maxWords > 0) {
```
Comment (AntoineRelief) [inline libs/shared/src/lib/survey/components/editor.ts:146]: @ronnyReliefapp 
Another issue
it only works when you edit the question, not when you load it:

CleanShot 2026-01-09 at 14 29 52


For example, if you open form, edit html and generate error on word count on purpose, wait for autosave to trigger, then close & reopen the form, the error isn't checked
